src/ros/ROSInterface.cpp

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright (c) 2023, Sascha Jongebloed
+ * All rights reserved.
+ *
+ * This file is part of KnowRob, please consult
+ * https://github.com/knowrob/knowrob for license details.
+ */
+
+#include "knowrob/ros/ROSInterface.h"
+// KnowRob
+#include "knowrob/knowrob.h"
+#include "knowrob/Logger.h"
+#include "knowrob/KnowledgeBase.h"
+#include "knowrob/queries/QueryParser.h"
+#include "knowrob/queries/AnswerError.h"
+#include "knowrob/modalities/BeliefModality.h"
+#include "knowrob/formulas/ModalFormula.h"
+#include "knowrob/modalities/KnowledgeModality.h"
+#include "knowrob/terms/ListTerm.h"
+// ROS
+#include <ros/ros.h>
+#include <ros/console.h>
+#include <ros/package.h>
+#include <iai_knowledge_msgs/GraphAnswer.h>
+#include <iai_knowledge_msgs/GraphQuery.h>
+#include <iai_knowledge_msgs/KeyValuePair.h>
+#include <iai_knowledge_msgs/askallAction.h>
+#include <boost/property_tree/json_parser.hpp>
+#include <utility>
+
+using namespace knowrob;
+
+ROSInterface::ROSInterface(const boost::property_tree::ptree& config)
+        : askall_action_server_(nh_, "knowrob/askall", boost::bind(&ROSInterface::executeAskAllCB, this, _1), false),
+          askone_action_server_(nh_, "knowrob/askone", boost::bind(&ROSInterface::executeAskOneCB, this, _1), false),
+          kb_(config)
+{
+    // Start all action servers
+    askall_action_server_.start();
+    askone_action_server_.start();
+    //askiterative_action_server_.start();
+}
+
+ROSInterface::~ROSInterface() = default;
+
+FormulaPtr
+ROSInterface::applyModality(const iai_knowledge_msgs::GraphQuery &query,
+                            FormulaPtr phi) {
+    FormulaPtr mFormula = std::move(phi);
```

Comment:
not sure about using std:move on shared pointer here. It might nullify `phi`! Probably better just to create a copy and to keep `phi` alive too. also you create a copy (of the shared pointer) whenever this function is called because you do not pass by reference. So maybe rather use `const FormulaPtr &phi` to avoid copy on calling the function